src/components/NewCommentForm.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import { Paper } from "@mui/material";
+import { postNewComment } from "../api";
+import { useParams } from "react-router-dom";
+
+const NewCommentForm = ({ comments, setComments }) => {
+  const { article_id } = useParams();
+  const [name, setName] = useState("");
+  const [body, setBody] = useState("");
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setComments(() => {
+      return [
+        { author: name, body: body, votes: 0, created_at: "Just now" },
+        ...comments,
+      ];
+    });
+    postNewComment(article_id, name, body).then((data) => {
+      setName("");
+      setBody("");
+    });
+  };
+
```

Comment:
Currently while the api request is going through the user is able to spam click the 'Submit' button, sending the request multiple times. If a user had a slow internet connection this might be a likely scenario- so we need to somehow block the user from submitting a comment while the request is being sent.

Reply:
Thanks, I'll fix this now